src/librustdoc/clean/inline.rs

```diff
@@ -200,7 +205,9 @@ pub(crate) fn record_extern_fqn(cx: &mut DocContext<'_>, did: DefId, kind: ItemT
             vec![crate_name, relative.last().expect("relative was empty")]
         }
     } else {
-        once(crate_name).chain(relative).collect()
+        once(crate_name)
+            .chain(cx.tcx.def_path_str(did).split("::").map(|entry| Symbol::intern(entry)).skip(1))
+            .collect()
```

Comment:
It actually returns the path used to import the item. Which doesn't work in case like:

```rust
pub mod quux {
    extern crate issue_30109_1 as bar;
    use self::bar::Bar;

    pub trait Foo {}

    // @has issue_30109/quux/trait.Foo.html \
    //          '//a/@href' '../issue_30109_1/struct.Bar.html'
    impl Foo for Bar {}
}
```

In here, `Bar` path will include `bar` but also `self` (in the form of the current crate name) whereas it definitely shouldn't.

Reply:
Here's the PR and feature which should allow to fix this definitely at some point: https://github.com/rust-lang/rust/pull/112853 (the PR doesn't fix it, just adds tests for the feature).